### PR TITLE
Add 3SF concepts

### PR DESCRIPTION
**Pull request by saltiniroberto**

### CODE REVIEW

File: reports/3sf.tex [hunk 1]
```diff
@@ -0,0 +1,77 @@
+%! TeX root = report.tex
+
+\section{Basic 3SF concepts}\label{sec:3sf}
+
+In this section, we summarize the basic concepts of the 3SF protocol that this project depends on.
+We refer the reader to the 3SF paper~\cite{d20243} for a more comprehensive explanation or any concepts that this project depends on which we might have missed listing here.
```
Comment (Kukovec) [inline reports/3sf.tex:6]:
```suggestion
We refer the reader to the 3SF paper~\cite{d20243} for a more comprehensive explanation.
```

File: reports/3sf.tex [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+\paragraph*{Validators.} Participants of the protocol are referred to as \emph{validators} with $n$ being their total number.
+Every validator is identified by a unique cryptographic identity and the public keys are common knowledge. 
+Each validator has a \emph{stake} but for the purpose of this project each validator's stake is set to 1.
+% If a validator $v_i$ misbehaves and a proof of this misbehavior is given, it loses a part of its stake ($v_i$ gets \emph{slashed}). 
+
+\paragraph*{Slots.} Time is divided into \emph{slots}. 
+In ideal condition a new block (see below) is expected to be proposed at the beginning of each slot.
```
Comment (Kukovec) [inline reports/3sf.tex:14]:
```suggestion
In ideal conditions a new block (see below) is expected to be proposed at the beginning of each slot.
```

File: reports/3sf.tex [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Given the definition above, each different block $B$ implicitly identifies a different finite \emph{chain} of blocks starting from block $B$, down to the genesis block, by recursively moving from a block to its parent.
+Hence, there is no real distinction between a block and the chain that it identifies.
+So, by chain $\chain$, we mean the chain identified by the block $\chain$.
+We write $\chain_1 \preceq \chain_2$ to mean that $\chain_1$ is a non-strict prefix of $\chain_2$.
+We say that $\chain_1$ \emph{conflicts} with $\chain_2$ if and only if neither $\chain_1 \preceq \chain_2$ nor $\chain_2 \preceq \chain_1$ holds. 
+
+\paragraph{Checkpoints.}
+In the protocol described in~\cite{d20243}, a \emph{checkpoint} is a tuple $(\chain, c)$, where \(\chain \) is a chain and \( c \) is a slot signifying where \( \chain \) is proposed for justification (this concept is introduced and explained below).
+However, for efficiency reasons, in the specification targeted by this project, a \emph{valid} checkpoint $\C$ is a triple $(H, c, p)$ where $H$ is the hash of a chain $\chain$, $c$ is as per the definition above and $p =  \chain.p$.
```
Comment (Kukovec) [inline reports/3sf.tex:30]:
```suggestion
However, for efficiency reasons, in the specification targeted by this project, a \emph{valid} checkpoint $\C$ is a triple $(H, c, p)$ where $H$ is the hash of a chain $\chain$, $c$ is the proposal slot as per the definition above and $p =  \chain.p$.
```

File: reports/3sf.tex [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+The total pre-order among checkpoints is defined:
+$\C \leq \C'$ if and only if either \(\C.c < \C'.c\) or, in the case where \(\C.c = \C'.c\), then \(\C.p \leq \C'.p\). 
+Also, $\C < \C'$ means $\C \leq \C' \land \C \neq \C'$.
+
+
+\paragraph*{FFG Votes.}
+Validators cast two main types of votes: \textsc{ffg-vote}s and \textsc{vote}s. 
+Each \textsc{vote} includes an \textsc{ffg-vote}.
+The specification targeted by this project only deals with \textsc{ffg-vote}s as the extra information included in  \textsc{vote}s has no impact on AccountableSafety.
+An \textsc{ffg-vote} is represented as a tuple \([\textsc{ffg-vote}, \C_1, \C_2, v_i]\), where {$v_i$ is the validator sending the \textsc{ffg-vote}\footnote{Digital signature are employed to ensure that $v_i$ is the actual sender and it is assumed that such digital signatures are unforgeable.}, while} \(\C_1\) and \(\C_2\) are checkpoints.
```
Comment (Kukovec) [inline reports/3sf.tex:40]:
```suggestion
An \textsc{ffg-vote} is represented as a tuple \([\textsc{ffg-vote}, \C_1, \C_2, v_i]\), where {$v_i$ is the validator sending the \textsc{ffg-vote}\footnote{Digital signatures are employed to ensure that $v_i$ is the actual sender and it is assumed that such digital signatures are unforgeable.}, while} \(\C_1\) and \(\C_2\) are checkpoints.
```

File: reports/3sf.tex [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+% Lastly, we say that a {chain \(\chain\)} \emph{is justified} if and only if there exists a justified checkpoint \(\C\) for which {\(\C.\chain = \chain\)}.
+
+% \paragraph*{Greatest justified checkpoint and chain.}
+% A checkpoint is considered justified in a view \(\V\) if \(\V\) contains a supermajority set of links justifying it. A \emph{justified} checkpoint which is maximal in \(\V\) with respect to the previously defined lexicographic ordering is referred to as the \emph{greatest justified checkpoint} in \(\V\), denoted as \(\GJ(\V)\). In the event of ties, they are broken arbitrarily. Chain \(\GJ(\V).\chain\) is referred to as the \emph{greatest justified chain}.
+
+\paragraph*{Finality.}
+A checkpoint \(\C\) is \emph{finalized} if it is justified and there exists a supermajority link with source \(\C\) and potentially different targets \(\C_i\) where \(\C_i.c = \C.c + 1\). A chain \(\chain\) is finalized if there exists a finalized checkpoint \(\C\) with \(\chain = \C.\chain\). The checkpoint \(\C = (B_\text{genesis}, 0)\) is finalized by definition.
```
Comment (Kukovec) [inline reports/3sf.tex:58]:
```suggestion
A checkpoint \(\C\) is \emph{finalized} if it is justified and there exists a supermajority link with source \(\C\) and potentially different targets \(\C_j\) where \(\C_j.c = \C.c + 1\). A chain \(\chain\) is finalized if there exists a finalized checkpoint \(\C\) with \(\chain = \C.\chain\). The checkpoint \(\C = (B_\text{genesis}, 0)\) is finalized by definition.
```
Just for consistency with the previous paragraph where C_j was the target.

File: reports/3sf.tex [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+% {Given a view $\V$, a finalized checkpoint which is maximal in \(\V\) with respect to the previously defined lexicographic ordering is referred to as the \emph{greatest finalized checkpoint} in \(\V\), denoted as \(\GF(\V)\).
+% In the event of ties, they are broken arbitrarily.
+% % We say that a chain 
+% We say that a chain $\chain$ is finalized according to a view $\V$ if and only if $\chain \preceq \GF(\V).\chain$.
+
+
+\paragraph*{Slashing.}
+A validator \(v_i\) is subject to slashing for sending two \emph{distinct} \textsc{ffg-vote}s \(\C_1 \to \C_2\) and \(\C_3 \to \C_4\) if either of the following conditions holds: {\(\mathbf{E_1}\) (Double voting)} if \(\C_2.c = \C_4.c\), implying that a validator must not cast distinct \textsc{ffg-vote}s for the same checkpoint slot; or {\(\mathbf{E_2}\) (Surround voting)} if \(\C_3 < \C_1\)
+ and \(\C_2.c < \C_4.c\), indicating that a validator must not vote using a lower checkpoint as source and must avoid voting within the span of its other votes.
```
Comment (Kukovec) [inline reports/3sf.tex:68]: Is surround voting defined with "<" or "<=" ?

**Review by Kukovec — APPROVED** [review]
LGTM, left some minor comments.

**Review by thpani — APPROVED** [review]
LGTM

File: reports/3sf.tex (hunk 6 above)
Comment (saltiniroberto) [reply reports/3sf.tex:68]: it is defined with "<"

File: reports/3sf.tex [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+\paragraph*{Slots.} Time is divided into \emph{slots}. 
+In ideal conditions a new block (see below) is expected to be proposed at the beginning of each slot.
+
+\paragraph{Blocks and Chains.} 
+A \emph{block} is a pair of elements, denoted as \( B = (b,p) \). Here, \( b \) represents the \emph{block body} -- essentially, the main content of the block which contains a batch of transactions grouped together.
+Each block body contains a reference pointing to its \emph{parent} block. 
+The second element of the pair, \( p \geq 0 \), indicates the \emph{slot} where the block \( B \) is proposed.
+By definition, if $B_p$ is the parent of $B$, then $B_p.p < B.p$.
+The \emph{genesis block} is the only block that does not have a parent {and has a negative slot}.
```
[the remark this answers is not in the document]
Comment (saltiniroberto) [reply reports/3sf.tex]: Yeah..that is how it is defined in the specs, but in the paper, we changed it to -1.
Given that this project is about the specs, I will change the text as you suggested.
Thanks.

File: reports/3sf.tex (hunk 3 above)
Comment (saltiniroberto) [reply reports/3sf.tex]: I have taken on your suggestion but changed the wording a bit.

**Pull request merged**